pkg/client/client.go

```diff
@@ -0,0 +1,195 @@
+	url := fmt.Sprintf("%s/%s", serverUrl, strings.TrimPrefix(path, "/"))
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating post request: %v", err)
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var data map[string]interface{}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %v", err)
+	}
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func Delete(ctx context.Context, c *http.Client, serverUrl string, path string) error {
+	url := fmt.Sprintf("%s/%s", serverUrl, strings.TrimPrefix(path, "/"))
+
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return fmt.Errorf("error creating delete request: %v", err)
+	}
+
+	resp, err := c.Do(req)
+	var data map[string]interface{}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %v", err)
+	}
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func Update(ctx context.Context, c *http.Client, serverUrl string, path string, body map[string]interface{}) error {
+	url := fmt.Sprintf("%s/%s", serverUrl, strings.TrimPrefix(path, "/"))
+
+	reqBody, err := json.Marshal(body)
+	if err != nil {
+		return fmt.Errorf("error marshalling JSON for request body: %v", err)
+	}
+
+	req, err := http.NewRequest("PATCH", url, strings.NewReader(string(reqBody)))
+	if err != nil {
+		return fmt.Errorf("error creating delete request: %v", err)
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+
+	var data map[string]interface{}
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %v", err)
+	}
+
+	err = json.Unmarshal(respBody, &data)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	err = CheckErrors(data)
+	if err != nil {
+		return err
+	}
+
+	return CheckErrors(data)
+}
+
+func CheckErrors(resp map[string]interface{}) error {
+	e, ok := resp["error"]
+	if ok {
+		return fmt.Errorf("returned errors, %v", e)
+	}
+}
+
+func createBase(ctx context.Context, r *api.Resource, serverUrl string, parameters map[string]string, suffix string) string {
+	serverUrl = strings.TrimSuffix(serverUrl, "/")
+	urlElems := []string{serverUrl}
+	for i, elem := range r.PatternElems {
+		if i == len(r.PatternElems)-1 {
+			continue
+		}
+
+		if i%2 == 0 {
+			urlElems = append(urlElems, elem)
+		} else {
+			paramName := elem[1 : len(elem)-1]
+			if value, ok := parameters[paramName]; ok {
```

Comment:
I think we should handle this error case, or return an empty string and handle it in the parent function.

I would opt toward the former since error types are more semantically clear, but I'm fine with either choice.